homework/index.js

```diff
@@ -30,15 +26,137 @@
+        text: repo.name,
+      });
+    });
+    return selectionElem;
+  }
+
+  function generateInfoSection(selected, data) {
+    const div = createAndAppend('div', container, { id: 'infoDiv' });
+    const selectedData = getSelectedData(selected, data);
+    createAndAppend('a', div, {
+      id: 'repoName',
+      text: `Repository: ${selectedData.name}`,
+      href: selectedData.html_url,
+      target: '_blank',
+    });
+    createAndAppend('p', div, {
+      id: 'desc',
+      text: `Description: ${selectedData.description}`,
+    });
+    createAndAppend('p', div, {
+      id: 'forks',
+      text: `Forks: ${selectedData.forks}`,
+    });
+    createAndAppend('p', div, {
+      id: 'updated_at',
+      text: `Updated: ${selectedData.updated_at}`,
+    });
+  }
+
+  async function generateContributorsSection(selected) {
+    const contributorsDiv = createAndAppend('div', container, { id: 'contsDiv' });
+    createAndAppend('p', contributorsDiv, { id: 'cont-header', text: 'Contributors' });
+    const contributorsLists = createAndAppend('ul', contributorsDiv, { id: 'conts-list' });
+
+    try {
+      const json = await fetchJSON(
+        `https://api.github.com/repos/HackYourFuture/${selected}/contributors`,
+      );
+      json.forEach(item => {
+        const contributorListItem = createAndAppend('li', contributorsLists, {
+          id: `${item.login}-list-item`,
+          class: 'cont-li',
+        });
+
+        createAndAppend('img', contributorListItem, {
+          id: `${item.login}Img`,
+          class: 'cont-avatar',
+          src: item.avatar_url,
+        });
+
+        const contributorDiv = createAndAppend('div', contributorListItem, {
+          id: `${item.login}Div`,
+        });
+
+        createAndAppend('a', contributorDiv, {
+          id: `${item.login}Name`,
+          class: 'cont-name',
+          text: item.login,
+          href: item.html_url,
+          target: '_blank',
+        });
+        createAndAppend('div', contributorDiv, {
+          id: `${item.login}Badge`,
+          class: 'cont-badge',
+          text: item.contributions,
+        });
+      });
+    } catch (error) {
+      createAndAppend('div', root, { text: error.message, class: 'alert-error' });
+    }
+  }
+
+  function updateContributorSection(selected) {
+    const contributorsDiv = document.getElementById('contsDiv');
+    contributorsDiv.parentNode.removeChild(contributorsDiv);
+    generateContributorsSection(selected);
+  }
+
+  function updateInfoSection(selected, data) {
+    const infoDiv = document.getElementById('infoDiv');
+    infoDiv.parentNode.removeChild(infoDiv);
+    generateInfoSection(selected, data);
+  }
+
+  function main(url) {
+    try {
+      const json = fetchJSON(url);
+      json.then(data => {
+        const repoNames = data
+          .map(repo => ({ name: repo.name, id: repo.id }))
+          .sort((a, b) => a.name.localeCompare(b.name));
+        generateSelections(repoNames);
```

Comment:
if you are returning the `selectionElem`, then probably you're having an extra getElementById in the next line.
and using it, what about:
```suggestion
        const selected = generateSelections(repoNames); // you may change the name of `selected` _(pastense)_ to `select` _(noun)_
```